pkg/policy/selectorcache.go

```diff
@@ -473,11 +518,17 @@ func (sc *SelectorCache) UpdateIdentities(added, deleted identity.IdentityMap, w
 				}
 			}
 			if len(dels)+len(adds) > 0 {
-				idSel.updateSelections()
+				updated = true
+				sc.selectorUpdates = sc.selectorUpdates.Append(idSel, txn)
+				idSel.updateSelections(txn)
 				idSel.notifyUsers(sc, adds, dels, wg)
 			}
 		}
 	}
+
+	if updated {
```

Comment:
Do we need to do anything to `txn` if `updated == false`? Rollback or such?

Reply:
No, as when there are no changes, there is no state to rollback and the txn can be left un-committed.